File: src/Application Specific/Classes/Entity Collections/StandardSentences/StandardSentence.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace StandardSentences

        Public Class StandardSentence

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub


#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "Standard Sentence Properties"

            Private _SentenceID As Integer = 0
            Public ReadOnly Property SentenceID() As Integer
                Get
                    Return _SentenceID
                End Get
            End Property
            Public WriteOnly Property SetSentenceID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SentenceID", Value)
                End Set
            End Property


            Private _Sentence As String = String.Empty
            Public ReadOnly Property Sentence() As String
                Get
                    Return _Sentence
                End Get
            End Property
            Public WriteOnly Property SetSentence() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Sentence", Value)
                End Set
            End Property

            Private _LastChanged As DateTime = Nothing
            Public Property LastChanged() As DateTime
                Get
                    Return _LastChanged
                End Get
                Set(ByVal Value As DateTime)
                    _LastChanged = Value
                End Set
            End Property

#End Region

        End Class

    End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Specific/Classes/Entity Collections/StandardSentences/StandardSentence.cs
using System;
using System.Collections;

namespace FSM.Entity
{
    namespace StandardSentences
    {
        public class StandardSentence
        {
            private DataTypeValidator _dataTypeValidator;

            public StandardSentence()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            private int _SentenceID = 0;

            public int SentenceID
            {
                get
                {
                    return _SentenceID;
                }
            }

            public object SetSentenceID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SentenceID", value);
                }
            }

            private string _Sentence = string.Empty;

            public string Sentence
            {
                get
                {
                    return _Sentence;
                }
            }

            public object SetSentence
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Sentence", value);
                }
            }
        }
    }
}